nginx.md

```diff
@@ -0,0 +1,190 @@
+	fastcgi_param PICO_URL_REWRITING 1;
+}
+```
+
+Please note that this is only provided as an **example**.  You should write your own PHP location block based on your personal system configuration.
+
+This `location` rule tells Nginx to send all pages ending in `.php` to an external php processor called `php-fpm`.  Again, setting this up is outside the scope of this document.  There are many tutorials available online.  Here is one for [Ubuntu 14.04](https://www.digitalocean.com/community/tutorials/how-to-install-linux-nginx-mysql-php-lemp-stack-on-ubuntu-14-04#3-install-php-for-processing).
+
+By default, `php-fpm` comes with a very insecure setting that can allow unauthorized code execution.  We've included a small `if` statement here that will protect you from this vulnerability.  If you've changed php-fpm's `cgi.fix_pathinfo` setting to `0`, you do not need this statement.
+
+The line `fastcgi_param PICO_URL_REWRITING 1;` informs Pico that we will be rewriting url's in Nginx.  This prevents Pico from using its own url-rewriting, and gives us nicer url's.
+
+## URL Rewriting
+
+There are several ways to approach Pico's url-rewriting in Nginx.  We'll be covering two different approaches in this document, but there really is no right answer.
+
+### Pico in Document Root
+
+If your Pico installation is in the `Document Root` of your website, then configuration is relatively easy.
+
+```
+location / {
+	try_files $uri $uri/ /?$uri&$args;
+}
+```
+
+This rule tells Nginx that whenever it looks up a url within your site, it should first look for a file of that name, then a directory, and finally, if neither exist, it should rewrite the url for Pico.  When the url is rewritten for Pico, it is formatted as `/` followed by a query string (starts with `?`) containing the file path (`$uri`) and then followed by any other query arguments (`$args`).  By referencing `/`, Nginx will load `index.php` (as defined in our server config above) and pass the rendering along to your php processor.
+
+### Pico in a Subfolder
+
+If your Pico installation is in a subfolder, configuration is slightly more complicated, but should be easy if you follow along.
+
+```
+location ~ ^/pico(.*) {
+	try_files $uri $uri/ /pico/?$1&$args;
+}
+```
+
+You'll notice that similar to our last example, we're sending Nginx to `index.php` with a query string of the url, but this time in a subfolder named `pico`.  What's different is that we're using `$1` to reference the page url instead of `$uri`.  This is because `$uri` will contain the entire url from the Document Root, but we only want the part that comes *after* the `/pico` subfolder.  Since we can't use `$uri` for this, we're using a Regular Expression (also called "regex") to determine the url for us.  This `location` rule looks for any url that starts with `/pico`, and takes note of whatever comes afterward.  In your own configuration, you'll need to replace `pico` on both lines with the location of your own subfolder.
```

Comment:
"What's different is...", "What is different is..." - shouldn't this be something like "The difference is..."?